plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2069,6 +2083,84 @@ public void testSplitPruningForFilterOnPartitionColumn()
+                        "REAL '1.0', " +
+                        "DOUBLE '1.0',  " +
+                        "CAST(1.0 as DECIMAL(5,2)),  " +
+                        "DATE '2021-07-24'," +
+                        "TIMESTAMP '2021-07-24 02:43:57.348000',  " +
+                        "TIMESTAMP '2021-07-24 03:43:57.348000'," +
+                        "TIMESTAMP '2021-07-24 04:43:57.348000' AT TIME ZONE 'America/Los_Angeles',  " +
+                        "'onefsadfdsf', " +
+                        "VARBINARY 'binary0/1values', " +
+                        "VARBINARY 'binary2/3values', " +
+                        "(CAST(ROW(null, 'this is a random value') AS ROW(int, varchar))), " +
+                        "array['uno', 'dos', 'tres'], " +
+                        "map(array[1,2], array['ek', 'one']))", 1);
+
+        @Language("SQL") String result = "" +
+                "SELECT" +
+                " true bool" +
+                ", 1 int_" +
+                ", 1 big" +
+                ", CAST('1.0' AS REAL) rl" +
+                ", CAST('1.0' AS DOUBLE) dbl" +
+                ", CAST('1.0' AS DECIMAL(5,2)) dec_" +
+                ", CAST('2021-07-24' AS DATE) dt" +
+                ", CAST('2021-07-24 02:43:57.348000' AS TIMESTAMP(6)) time_" +
+                ", CAST('2021-07-24 03:43:57.348000' AS TIMESTAMP(6)) timestamp_" +
+                ", 'onefsadfdsf' string_";
+        assertQuery("SELECT bool, int_, big, rl, dbl, dec_, dt, time_, timestamp_, string_ FROM test_all_types", result);
+        query("SELECT timestamptz_, fixed, binary_, row_, arr, mp FROM test_all_types").assertThat()
+                .matches(
+                        resultBuilder(
+                                SESSION,
+                                TIMESTAMP_TZ_MICROS,
+                                VARBINARY,
+                                VARBINARY,
+                                RowType.from(ImmutableList.of(new RowType.Field(Optional.of("id"), INTEGER), new RowType.Field(Optional.of("vc"), VARCHAR))),
+                                new ArrayType(VARCHAR),
+                                new MapType(INTEGER, VARCHAR, new TypeOperators()))
+                                .row(
+                                        ZonedDateTime.of(2021, 7, 23, 15, 43, 57, 348000000, ZoneId.of("UTC")),
+                                        "binary0/1values".getBytes(StandardCharsets.UTF_8),
+                                        "binary2/3values".getBytes(StandardCharsets.UTF_8),
+                                        new MaterializedRow(Arrays.asList(null, "this is a random value")),
+                                        Arrays.asList("uno", "dos", "tres"),
+                                        new HashMap<Integer, String>()
+                                        {{
+                                            put(1, "ek");
+                                            put(2, "one");
+                                        }})
```

Comment:
I did not know you can do that in java :)
Anyway why not just `ImmutableMap.of(...)` as usual?

Reply:
I was sure assertion would fail but this is not the case. Thanks